alacritty/src/config/monitor.rs

```diff
@@ -15,18 +14,34 @@ const DEBOUNCE_DELAY: Duration = Duration::from_millis(10);
 #[cfg(not(any(target_os = "linux", target_os = "macos", target_os = "windows")))]
 const DEBOUNCE_DELAY: Duration = Duration::from_millis(1000);
 
-pub fn watch(mut paths: Vec<PathBuf>, event_proxy: EventProxy) {
-    // Canonicalize all paths, filtering out the ones that do not exist.
-    paths = paths
-        .drain(..)
-        .filter_map(|path| match fs::canonicalize(&path) {
-            Ok(path) => Some(path),
-            Err(err) => {
-                error!("Unable to canonicalize config path {:?}: {}", path, err);
-                None
-            },
-        })
-        .collect();
+pub fn watch(paths: Vec<PathBuf>, event_proxy: EventProxy) {
+    let paths = {
+        let mut paths_tmp = Vec::new();
+        for path in paths {
+            if path.exists() {
+                paths_tmp.push(path.clone());
+
+                // Also watch the targets of symbolic links
+                match path.symlink_metadata() {
+                    Ok(metadata) if metadata.file_type().is_symlink() => {
+                        match path.canonicalize() {
```

Comment:
By only canonicalizing paths that are symlinks, we're leaving us open for paths to be watched multiple times even though they're not symlinks.

Take the following example:
 - /tmp/../alacritty.yml
 - /alacritty.yml

Both are the same file, but only one of them needs to be watched.

Ultimately this should never happen without user error, but doing it "right" is basically the same effort as not, so I see no reason why we wouldn't do it properly.

Reply:
I couldn't find a function in the standard library that canonicalises paths without resolving symlinks. What are you thinking?

Reply:
We can still check if the file is a symlink, that is no problem. But we should **always** push the canonicalized path and only push the non-canonicalized one if the file is a symlink. Basically exactly the opposite way of what you've done right now.

Reply:
Indeed, that makes more sense.